napari/_qt/widgets/qt_keyboard_settings.py

```diff
@@ -406,7 +406,8 @@ def _set_keybinding(self, row, col):
             )
             for mod in {"Shift", "Ctrl", "Alt", "Cmd", "Super", 'Meta'}:
                 # we want to prevent multiple modifiers but still allow single modifiers.
-                if new_shortcut.endswith('-' + mod):
+                separator = '-' if sys.platform == 'darwin' else '+'
+                if new_shortcut.endswith(f'{separator}{mod}'):
```

Comment:
Note: While I was checking the code for the `-` usage and although I did a change here looking for consistency, I noticed that maybe this logic is never triggered due to the modifiers handling on the [`EditorWidget`](https://github.com/napari/napari/blob/2f6c7073709a10e1e46e113e50ca298dfadb715f/napari/_qt/widgets/qt_keyboard_settings.py#L564). I think the editor widget doesn't allow you to only set modifier keys as shortcuts :thinking: If you run from latest main (and least on Windows) you can see the following: 
 
![shortcuts_only_mod](https://github.com/napari/napari/assets/16781833/f46bebc6-e20f-4b3d-b4e8-75e671b09677)

The keypress updates the text but it never sets the shortcut unless another key (different from the modifier keys) is press

Reply:
[7dfd87f](https://github.com/napari/napari/pull/6330/commits/7dfd87fbb2fd96cb9fc9f38f28ee5b5a9b85dd42) should fix the above (if I'm understanding correctly that the expected behavior is to be able to set a single modifier key as a shortcut while marking as invalid the usage of multiple modifiers keys without any other keys)

Reply:
Oh I see now, I will give this another check then :+1: